src/sage/manifolds/chart.py

```diff
@@ -3416,7 +3520,7 @@ def _latex_(self):
         """
         return latex(self._chart1) + r' \rightarrow ' + latex(self._chart2)
 
-    def __eq__(self, other):
+    def __eq__(self, other: CoordChange) -> bool:
```

Comment:
Not correct. It is allowed to pass an object of a different type as ``other``. The method returns ``False`` in this case.

Reply:
There seems to be two conventions for this. Either you use indeed `object` for `other`, or you restrict `other` to some meaningful subset. The later is employed by e.g [numpy](https://github.com/numpy/numpy/blob/877522813effd8b052a3a3db8a51e5bad93d7540/numpy/array_api/_array_object.py#L495) and [many others](https://github.com/search?q=__eq__+%22other%3A+union%22&type=code). The logic here seems to be that its more helpful to have your static checker complain that you compare apples and bananas, then to correctly handle the relatively rare case of comparing two completely different object (in which case you can ignore the error).